src/storage/transaction/TossEdgeIterator.h

```diff
@@ -0,0 +1,224 @@
+                    lastDstId_ = NebulaKeyUtils::
+                                    getDstId(planContext_->vIdLen_, iter_->key()).str();
+                    lastIsLock_ = false;
+                    if (stopAtFirstEdge_) {
+                        stopSearching_ = true;
+                    }
+                    return;
+                } else if (lastIsLock_) {
+                    LOG_IF(INFO, FLAGS_trace_toss) << "tempory set val for edge ver=" << ver;
+                    auto tryLockData = recoverEdges_.back()->tryWLock();
+                    if (tryLockData && tryLockData->first.empty()) {
+                        tryLockData->first = iter_->key().str();
+                        tryLockData->second = iter_->val().str();
+                    }
+                    lastIsLock_ = false;
+                }
+                continue;
+            } else if (NebulaKeyUtils::isLock(planContext_->vIdLen_, iter_->key())) {
+                std::string rawKey = NebulaKeyUtils::toEdgeKey(iter_->key());
+                auto rank = NebulaKeyUtils::getRank(planContext_->vIdLen_, rawKey);
+                auto dstId = NebulaKeyUtils::getDstId(planContext_->vIdLen_, rawKey).str();
+                if (!lastIsLock_ && rank == lastRank_ && dstId == lastDstId_) {
+                    continue;
+                }
+                if (rank != lastRank_ || dstId != lastDstId_) {
+                    recoverEdges_.emplace_back(
+                        std::make_shared<folly::Synchronized<KV>>(std::make_pair("", "")));
+                }
+
+                resumeTasks_.emplace_back(
+                    planContext_->env_->txnMan_->resumeTransaction(planContext_->vIdLen_,
+                                                                   planContext_->spaceId_,
+                                                                   iter_->key().str(),
+                                                                   iter_->val().str(),
+                                                                   recoverEdges_.back()));
+                lastRank_ = NebulaKeyUtils::getRank(planContext_->vIdLen_, rawKey);
+                lastDstId_ = NebulaKeyUtils::getDstId(planContext_->vIdLen_, rawKey).str();
+                lastIsLock_ = true;
+                continue;
+            } else {
+                LOG_IF(INFO, FLAGS_trace_toss)
+                    << "next to a weird record: \n"
+                    << folly::hexDump(iter_->key().data(), iter_->key().size())
+                    << folly::hexDump(iter_->val().data(), iter_->val().size());
+            }
+        }
+        LOG_IF(INFO, FLAGS_trace_toss) << "no more iter to read";
+
+        if (needWaitResumeTask_) {
+            folly::collectAll(resumeTasks_).wait();
+            needWaitResumeTask_ = false;
+            recoverEdgesIter_ = recoverEdges_.begin();
+            firstLoop_ = true;
+        }
+
+        while (recoverEdgesIter_ != recoverEdges_.end()) {
```

Comment:
We'd beeter add a init value for `recoverEdgesIter_` in ctor

Reply:
good catch. fixed